webui/react/src/components/UserSettingsModal.tsx

```diff
@@ -0,0 +1,88 @@
+import { useObservable } from 'micro-observables';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { Modal } from 'components/kit/Modal';
+import userSettings from 'stores/userSettings';
+import { isObject } from 'utils/data';
+import handleError from 'utils/error';
+import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
+
+import CodeEditor from './kit/CodeEditor';
+
+interface Props {
+  onSave?: () => void;
+}
+
+const UserSettingsModal: React.FC<Props> = ({ onSave }: Props) => {
+  const loadableState = useObservable(userSettings.getAll());
+  const state = Loadable.getOrElse(undefined, loadableState);
```

Comment:
While settings are loading you're showing an empty editor which is incorrect. You should be showing loading state.

Reply:
Alright, this should be fixed. While the settings are loading it will show CodeEditor's loading state.
![Screenshot 2023-07-13 at 12-21-42 Home - Determined](https://github.com/determined-ai/determined/assets/15078396/6be22596-83fa-44fe-b80b-77c2a34e9a7f)